src/Middleware/Spa/SpaServices.Extensions/src/Proxying/SpaProxy.cs

```diff
@@ -29,6 +29,9 @@ internal static class SpaProxy
         // Others just aren't applicable in proxy scenarios
         private static readonly string[] NotForwardedWebSocketHeaders = new[] { "Accept", "Connection", "Host", "User-Agent", "Upgrade", "Sec-WebSocket-Key", "Sec-WebSocket-Protocol", "Sec-WebSocket-Version" };
 
+		// In case the connection to the client is HTTP/2 or HTTP/3 and to the server HTTP/1.1 or less, let's get rid of the HTTP/1.1 only headers
+		private static readonly string[] InvalidH2H3Headers = new[] { "Connection", "Transfer-Encoding", "Keep-Alive", "Upgrade", "Proxy-Connection" };
```

Comment:
Better as a hash table?

Reply:
That makes sense. I was (wrongly) assuming, that this code would anyway only be in use in development.
I used the opportunity to also change the other header name lists to `HaseSet`s.